File: typescript/medium/ways-to-express-an-integer-as-sum-of-powers.ts
function numberOfWays(n: number, x: number): number {
    const MOD = 1_000_000_007;
    const dp: number[] = new Array(n + 1).fill(0);
    dp[0] = 1;

    for (let i = 1; i <= n; i++) {
        const val = Math.pow(i, x);
        if (val > n) break;
        for (let j = n; j >= val; j--) {
            dp[j] = (dp[j] + dp[j - val]) % MOD;
        }
    }

    return dp[n];
}

File: csharp/medium/ways-to-express-an-integer-as-sum-of-powers.cs
int NumberOfWays(int n, int x)
    {
        const int MOD = 1_000_000_007;
        int[] dp = new int[n + 1];
        dp[0] = 1;

        for (int i = 1; i <= n; i++)
        {
            long val = (long)Math.Pow(i, x);
            if (val > n) break;
            for (int j = n; j >= val; j--)
            {
                dp[j] = (int)((dp[j] + dp[j - (int)val]) % MOD);
            }
        }

        return dp[n];
    }
